lib/Conversion/Utils.h

```diff
@@ -6,6 +6,11 @@
 namespace mlir {
 namespace heir {
 
+// Adds conversion patterns that deal with tensor<..xsource_type>
+// when source_type will be type converted to tensor<...>, too
+void addTensorPatterns(TypeConverter &typeConverter,
```

Comment:
While the comment is helpful, I think a more explicit name like `populateTensorOfTensorWorkaroundPatterns` would be even better.

Reply:
I went with `addTensorOfTensorConversionPatterns` for now, to match the naming with the existing `addStructuralConversionPatterns`